src/metabase/driver/impl.clj

```diff
@@ -206,5 +206,5 @@
         (when-not (initialized? driver)
           (log/info (u/format-color 'yellow (trs "Initializing driver {0}..." driver)))
           (log/debug (trs "Reason:") (u/pprint-to-str 'blue (drop 5 (u/filtered-stacktrace (Thread/currentThread)))))
-          (swap! initialized-drivers conj driver)
-          (init-fn driver))))))
+          (init-fn driver)
+          (swap! initialized-drivers conj driver))))))
```

Comment:
this is the fix for the race condition. Don't add `driver` to the set of `initialized-drivers` until after we finish initializing it